tests/ui/async-await/lifetime-name-annotation-2021-ice-123420.rs

```diff
@@ -0,0 +1,18 @@
+// issue rust-lang/rust#123420
+// ICE assertion failed: i32 != &'{erased} mut i32
+//@ edition:2021
+#![feature(async_closure)]
```

Comment:
this should go into `async-await/async-closures`

Reply:
also, i don't think we even need this test. I added enough tests in #123660.

Reply:
ah neat; I'll boot this one then :)